WooCommerce/Classes/ViewRelated/Orders/OrderStatusListViewController.swift

```diff
@@ -0,0 +1,249 @@
+
+        let orderID = order.orderID
+        let undoStatus = order.statusKey
+
+        let done = updateOrderAction(siteID: order.siteID, orderID: orderID, statusKey: newStatus)
+        let undo = updateOrderAction(siteID: order.siteID, orderID: orderID, statusKey: undoStatus)
+
+        StoresManager.shared.dispatch(done)
+
+        displayOrderUpdatedNotice {
+            StoresManager.shared.dispatch(undo)
+        }
+    }
+
+    /// Returns an Order Update Action that will result in the specified Order Status updated accordingly.
+    ///
+    private func updateOrderAction(siteID: Int, orderID: Int, statusKey: String) -> Action {
+        return OrderAction.updateOrder(siteID: siteID, orderID: orderID, statusKey: statusKey, onCompletion: { error in
+            guard let error = error else {
+                WooAnalytics.shared.track(.orderStatusChangeSuccess)
+                return
+            }
+
+            WooAnalytics.shared.track(.orderStatusChangeFailed, withError: error)
+            DDLogError("⛔️ Order Update Failure: [\(orderID).status = \(statusKey)]. Error: \(error)")
+
+            self.displayErrorNotice(orderID: orderID)
+        })
+    }
+}
+
+
+/// MARK: - UITableViewDatasource conformance
+extension OrderStatusListViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return statusResultsController.sections[section].numberOfObjects
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: StatusListTableViewCell.reuseIdentifier,
+                                                       for: indexPath) as? StatusListTableViewCell else {
+            fatalError()
+        }
+
+        let status = statusResultsController.object(at: indexPath)
+        cell.textLabel?.text = status.name
+
+        return cell
+    }
+}
+
+
+/// MARK: - UITableViewDelegate conformance
+extension OrderStatusListViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        selectedStatus = statusResultsController.object(at: indexPath)
+    }
+}
+
+
+/// MARK: - Error handling
+extension OrderStatusListViewController {
+    /// Displays the `Order Updated` Notice. Whenever the `Undo` button gets pressed, we'll execute the `onUndoAction` closure.
+    ///
+    private func displayOrderUpdatedNotice(onUndoAction: @escaping () -> Void) {
+        let message = NSLocalizedString("Order status updated", comment: "Order status update success notice")
+        let actionTitle = NSLocalizedString("Undo", comment: "Undo Action")
+        let notice = Notice(title: message, feedbackType: .success, actionTitle: actionTitle, actionHandler: onUndoAction)
+
+        AppDelegate.shared.noticePresenter.enqueue(notice: notice)
+    }
```

Comment:
Woo code styling tip: add some breathing room to your code. Leave a newline between an ending curly brace and a new comment.

```
}

/// Comment
///
func display()
```